zrml/swaps/src/lib.rs

```diff
@@ -733,5 +736,38 @@ mod pallet {
 
             Ok(next_pool_id)
         }
+
+        fn pool(pool_id: PoolId) -> Result<Pool<Self::Balance, Self::MarketId>, DispatchError> {
+            Ok(Self::pool_by_id(pool_id)?)
+        }
+
+        fn set_pool_as_stale(
+            market_type: &MarketType,
+            pool_id: PoolId,
+            outcome_report: &OutcomeReport,
+        ) -> Result<(), DispatchError> {
+            if let MarketType::Categorical(_) = market_type {
+                if let &OutcomeReport::Categorical(winning_asset_idx) = outcome_report {
+                    <Pools<T>>::try_mutate(pool_id, |pool| {
+                        let pool_mut = if let Some(el) = pool {
+                            el
+                        } else {
+                            return Err(Error::<T>::PoolDoesNotExist.into());
+                        };
+                        pool_mut.assets.retain(|el| {
+                            if let &Asset::CategoricalOutcome(_, idx) = el {
+                                idx == winning_asset_idx
+                            } else {
+                                false
+                            }
+                        });
```

Comment:
```suggestion
                        pool_mut.assets.retain(|el| {
                            if let &Asset::CategoricalOutcome(_, idx) = el {
                                idx == winning_asset_idx
                            } else if let &Asset::Ztg = el {
                              true
                            } else {
                                false
                            }
                        });
```

Ztg should be retained too